app_dart/lib/src/request_handling/authentication.dart

```diff
@@ -180,11 +185,81 @@ class AuthenticationProvider {
       } finally {
         client.close();
       }
+    } else if (accessToken != null) {
+      return authenticateAccessToken(accessToken: accessToken, clientContext: clientContext, log: log);
     } else {
       throw const Unauthenticated('User is not signed in');
     }
   }
 
+  /// Verify [accessToken] with Google Auth, and return an [AuthenticatedContext] if
+  /// it is verified to have access to the Cocoon backend.
+  /// 
+  /// Otherwise, throw [Unauthenticated] exception.
+  Future<AuthenticatedContext> authenticateAccessToken({
+    @required String accessToken,
+    @required ClientContext clientContext,
+    @required Logging log,
+  }) async {
+    final HttpClient client = _httpClientProvider();
+    try {
+      final Uri googleOauthTokenInfoUri = Uri.https(
+        'www.googleapis.com',
+        '/oauth2/v1/tokeninfo',
+        <String, String>{
+          'access_token': accessToken,
+        },
+      );
+      final HttpClientRequest verifyAccessTokenRequest = await client.getUrl(googleOauthTokenInfoUri);
+      final HttpClientResponse verifyAccessTokenResponse = await verifyAccessTokenRequest.close();
+
+      /// This also checks for when an access token is expired.
+      if (verifyAccessTokenResponse.statusCode != HttpStatus.ok) {
+        final String body = await utf8.decodeStream(verifyAccessTokenResponse);
+        log.warning('Access token verification failed: ${verifyAccessTokenResponse.statusCode}; $body');
```

Comment:
I suspect this will be really noisy. Will this log be helpful for us?

Reply:
This is mirrored from the id token authentication to maintain consistency